src/shared/clients/ec2Client.ts

```diff
@@ -30,8 +32,27 @@ export class Ec2Client {
             'NextToken',
             'Reservations'
         )
-        const instances = this.getInstancesFromReservations(collection)
-        return instances
+        const extractedInstances = this.getInstancesFromReservations(collection)
+        const instancesWithStatuses = await this.addStatusesToInstances(extractedInstances)
+        const instancesWithNames = await this.addNamesToInstances(instancesWithStatuses)
+
+        return instancesWithNames
+    }
+
+    protected async addStatusesToInstances(
```

Comment:
-  `addStatusesToInstances` and `addNamesToInstances` are separate functions, but could they be combined? is there a case where we would want to "update statues" but not "update names"? Combining them helps discoverability and readability.
- "add" as the verb confused me because I expected to see something being appended/inserted/concat'd, but actually we're updating-in-place.

```suggestion
    protected async updateStatuses(
```

Reply:
No use case that I can think of at least right now of wanting them to be separated. I definitely see how the `add` language is confusing here. I combined them into `updateInstancesDetail`, which does both the adding of status and name.